design/secrets_provider_milestone1_2_design.md

```diff
@@ -0,0 +1,380 @@
+
+We chose to use the pub/sub pattern over waiting for a configurable amount of time because this delivers a nicer customer experience. The customer's K8s Secrets will be updated immediately which in turn avoids situations where customer applications fail to connect to their endpoints. 
+
+#### Flow
+
+1. Configure K8s Secrets with ***conjur-map\*** metadata defined 
+
+   - *Optional:* add label to K8s Secret 
+
+2. Create Service Account for Secrets Provider 
+
+3. Create Role / ClusterRole for the Service Accounts with **get**/**update** privileges
+
+4. Create RoleBinding / ClusterRoleBinding and bind the Service Accounts to the role from previous step
+
+5. Create/Deploy Secrets Providers as Pods in deployment manifest(s) **(\*Milestone 2)\***
+
+6. - If label filtering is defined, add "K8S_SECRET_LABEL" in the Secrets Provider manifest(s)
+
+7. When the Secrets Provider receives a message that Conjur secrets have changed, the process of fetching and updating the K8s Secret(s) repeats
+
+
+
+#### Customer experience
+
+Because the Secrets Provider needs to now update K8s Secrets with Conjur Secrets continuously, it can no longer run as a Job and will require that the customer deploy the Secrets Provider as a Pod (kind: Pod instead of kind: Job)
+
+```yaml
+apiVersion: app/v1
+kind: Pod
+metadata:
+  name: secrets-provider
+spec:
+  template:
+    spec:
+      serviceAccountName: prod-sa
+      containers:
+      - image: secrets-provider:latest
+        name: cyberark-secrets-provider-1
+....
+```
+
+
+
+#### Code changes
+
+Add publishing mechanism to Conjur server. Work TBD.
+
+
+
+### Order of Deployment
+
+##### Milestone 1 and 2 
+
+To ensure that on first run our Secrets Provider runs first, we will request from the customer follow the following setup order:
+
+1. Add all necessary K8s Secrets and their labels to the Secrets Provider manifest
+
+2. Run the Secrets Provider
+
+3. Run application pods
+
+   
+
+### Lifecycle/Deletion
+
+##### Milestone 1
+
+The lifecycle of the Job is the length of the process. In other words, the time it takes to retrieve the Conjur secrets value and update them in the K8s Secrets. The customer can configure the Job to stay "alive" by adding **ttlSecondsAfterFinished**, with the desired about of seconds, to their Secrets Provider yaml. Because of the nature of a Job, a manual delete is not necessary.
+
+##### Milestone 2
+
+Because we are upgrading to a Pod that does not terminate upon task completion in Milestone 2, if the customer wants to stop the Secrets Provider process, they will need to delete the pod.
+
+API endpoint for deleting Pod:
+
+```yaml
+delete /api/v1/namespaces/{namespace}/pods/{name}
+```
+
+### Backwards compatibility
```

Comment:
please address also backwards compatibility from milestone 1 to the current implementation